src/Tools/BuildValidator/CompilationDiff.cs

```diff
@@ -114,124 +131,183 @@ private CompilationDiff(ImmutableArray<Diagnostic> diagnostics, string originalP
                         var debugPath = options.DebugPath;
                         logger.LogInformation($@"Writing diffs to ""{Path.GetFullPath(debugPath)}""");
 
-                        var assemblyName = Path.GetFileNameWithoutExtension(originalBinaryPath.Name);
-                        var assemblyDebugPath = Path.Combine(debugPath, assemblyName);
+                        fixed (byte* ptr = rebuildBytes)
```

Comment:
It may be easier to review this section by just looking at the new code vs. the diff. This essentially attempts to streamline how we write out build artifacts. I wanted to have a single path for writing build artifacts that we just call into twice.